### PR TITLE
Preview metrics package

### PR DESCRIPTION
**Pull request by victlu**
Use preview System.Diagnostic.DiagnosticSource package from DotNet team.

This is branch off of PR #2053

### CODE REVIEW

File: build/Common.props [hunk 1]
```diff
@@ -62,6 +62,11 @@
     </PackageReference>-->
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (cijothomas) [inline build/Common.props:65]: could you move this to non.prod.props so only tests get it.
https://github.com/open-telemetry/opentelemetry-dotnet/pull/2057/files#diff-5e21151922a3cc1af4e89f73a993199997928b4299905fcedc5a9de10d9294d5R18

File: src/System.Diagnostics.Metrics.Temp/Counter.cs [hunk 2]
```diff
@@ -92,7 +92,7 @@ public void Add(T delta)
             T delta,
             params KeyValuePair<string, object?>[] tags)
         {
-            this.RecordMeasurement(delta, tags);
+            this.RecordMeasurement(delta, new ReadOnlySpan<KeyValuePair<string, object?>>(tags));
```
Comment (cijothomas) [inline src/System.Diagnostics.Metrics.Temp/Counter.cs:95]: we'll need to delete this whole file, right?

File: src/System.Diagnostics.Metrics.Temp/Instrument.cs [hunk 3]
```diff
@@ -72,6 +73,14 @@ protected Instrument(Meter meter, string name, string? unit, string? description
         /// </summary>
         public virtual bool IsObservable => false;
 
+        public void Register(MeterListener listener)
```
Comment (cijothomas) [inline src/System.Diagnostics.Metrics.Temp/Instrument.cs:76]: we need to delete this whole file?

**Review by cijothomas — CHANGES_REQUESTED** [review]
https://github.com/open-telemetry/opentelemetry-dotnet/pull/2057/files This PR does not face the assembly load issue, so you may model the props file after it. I have found a test failure in .NET6. While its being investigated separately, you can Ignore that test in metrics branch to unblock.

Also there are many leftover files in the temp project, please nuke them too.

**Review by victlu — COMMENTED** [review]
One test is failing.

### Windows
```
  Failed OpenTelemetry.Trace.Tests.TracerProviderSdkTest.TracerProviderSdkInvokesSamplingWithCorrectParameters [29 ms]
  Error Message:
   Assert.Null() Failure
Expected: (null)
Actual:   Activity { ActivityTraceFlags = Recorded, Baggage = [], Context = System.Diagnostics.ActivityContext, DisplayName = "customContext", Duration = 00:00:00, ... }
  Stack Trace:
     at OpenTelemetry.Trace.Tests.TracerProviderSdkTest.TracerProviderSdkInvokesSamplingWithCorrectParameters() in D:\a\opentelemetry-dotnet\opentelemetry-dotnet\test\OpenTelemetry.Tests\Trace\TracerProviderSdkTest.cs:line 40
```

### Linux
```
  Failed OpenTelemetry.Trace.Tests.TracerProviderSdkTest.TracerProviderSdkInvokesSamplingWithCorrectParameters [8 ms]
  Error Message:
   Assert.Null() Failure
Expected: (null)
Actual:   Activity { ActivityTraceFlags = Recorded, Baggage = [], Context = System.Diagnostics.ActivityContext, DisplayName = "customContext", Duration = 00:00:00, ... }
  Stack Trace:
     at OpenTelemetry.Trace.Tests.TracerProviderSdkTest.TracerProviderSdkInvokesSamplingWithCorrectParameters() in /home/runner/work/opentelemetry-dotnet/opentelemetry-dotnet/test/OpenTelemetry.Tests/Trace/TracerProviderSdkTest.cs:line 107
```

**Pull request closed without merging**